File: backend/Models/SalesItem.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace BonelessPharmacyBackend
{
    /// <summary>
    /// A product in the BonelessPharmacy system
    /// </summary>
    public class SalesItem
    {
        [Key]
        /// <summary>
        /// The primary key of the SalesItem
        /// </summary>
        /// <returns></returns>
        public int Id { get; set; }

        [Required]
        /// <summary>
        /// The name
        /// </summary>
        /// <example>
        /// Panadol Paracetamol
        /// </example>
        /// <returns></returns>
        public string Name { get; set; }

        /// <summary>
        /// The supplier ID number for the product
        /// </summary>
        /// <remarks>
        /// This may get overriden with each order if the supplier changes
        /// </remarks>
        /// <returns></returns>
        public string SupplierCode { get; set; }

        [Required]
        /// <summary>
        /// The price for the product, GST excluded
        /// </summary>
        /// <example>
        /// 23.60
        /// </example>
        /// <returns></returns>
        public double Price { get; set; }

        [Required]
        /// <summary>
        /// The stock of this item in store
        /// </summary>
        /// <example>
        /// 16
        /// </example>
        /// <returns></returns>
        public uint StockOnHand { get; set; }

        [Required]
        /// <summary>
        /// How much individual parts of the item are in a single sale of it.!--
        /// This can be a measurement of count, volume, size etc...
        /// </summary>
        /// <example>
        /// 10
        /// </example>
        /// <remarks>
        /// Maybe use "Size" instead?
        /// </remarks>
        /// <returns></returns>
        public int Amount { get; set; }
    }
}

File: frontend/app/ts/Models/SalesItem.ts
interface SalesItem {
    /**
     * The primary key of the SalesItem
     */
    id: number;

    /**
     * The name of the item
     * @example Panadol Paracetamol
     */
    name: string;

    /**
     * The supplier ID number for the product
     */
    supplierCode: string;

    /**
     * The price of the product, GST excluded
     * @example 23.60
     */
    price: number;

    /**
     * The stock of this item in store
     * @example 16
     */
    stockOnHand: number;

    /**
     * How much individual parts of the item are in a single sale of it.
     * This can be a measurement of count, volume, size etc...
     */
    amount: number;

    /**
     * The id of the sales item measurement
     */
    measurementId: number;

    /**
     * The related measurement object
     */
    measurement: Measurement;

    /**
     * Probs don't need this
     * 0 = No, 1 = Yes
     */
    isArchived: number;
}
